cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -492,9 +512,20 @@ private MongoDatabase getDatabase(String dbName) {
 
         if (client == null) {
             if (mongoUsername.length() != 0) {
-                MongoCredential credential = MongoCredential.createCredential(mongoUsername, dbName,
+                String authSource;
+                if ((mongoAuthSource != null) && !mongoAuthSource.isEmpty()) {
+                    authSource = mongoAuthSource;
+                } else {
+                    authSource = dbName;
```

Comment:
Not sure if the dbName is the right authSource in case it is not specified.

Reply:
Before this modification dbName was the value used in the previous version. If authSource is not configured then the behavior of the code is the same as before maintaining the backward compatibility. From my point of view it is better to keep the code as it is and not setting authSource as a constant to "admin".

Reply:
NTC

Reply:
Yes, dbName is one thing and authSource is another. The parameter dbName is the name of the database to create and based on "Fiware-Service" where the collections will be stored. And authSource is used to specify a user DB where the credentials are defined, as documented by Mongo and for the credentials objects. These are 2 different things.
dbName is used in client.getDatabase(dbName) (class MongoClient) and authSource is used by credential object (MongoCredential) if only there is a need of authentication.